scripts/dasharo-deploy

```diff
@@ -367,87 +628,224 @@ blob_transmission() {
-    check_flash_lock
-    verify_artifacts bios
+    # Following command will reset device, so the function will not quit:
+    $DASHARO_ECTOOL flash "$EC_UPDATE_FILE" &>> $ERR_LOG_FILE
+    error_check "Failed to update EC firmware"
 
-    check_intel_regions
-    check_blobs_in_binary $BIOS_UPDATE_FILE
-    check_if_me_disabled
-    set_intel_regions_update_params "-N --ifd -i bios"
-  fi
+    return 0
+  elif [ "$_mode" == "install" ]; then
 
-  if [ "$HAVE_EC" == "true" ]; then
-    echo "Checking for Open Source Embedded Controller firmware"
-    $DASHARO_ECTOOL info >> $ERR_LOG_FILE 2>&1
-    if [ $? -eq 0 ]; then
+    if check_for_opensource_firmware; then
       echo "Device has already Open Source Embedded Controller firmware, do not flash EC..."
     else
-      _ec_fw_version=$($FLASHROM -p "$PROGRAMMER_EC" ${FLASH_CHIP_SELECT} | grep "Mainboard EC Version" | tr -d ' ' | cut -d ':' -f 2)
+      _ec_fw_version=$($FLASHROM get_ec_firm_version_mock check -p "$PROGRAMMER_EC" ${FLASH_CHIP_SELECT} | grep "Mainboard EC Version" | tr -d ' ' | cut -d ':' -f 2)
+
       if [ "$_ec_fw_version" != "$COMPATIBLE_EC_FW_VERSION" ]; then
-        print_warning "EC version: $_ec_fw_version is not supported, update required"
-        install_ec
+        echo "Installing EC..."
+        $FLASHROM -p "$PROGRAMMER_EC" ${FLASH_CHIP_SELECT} -w "$EC_UPDATE_FILE" >> $FLASHROM_LOG_FILE 2>> $ERR_LOG_FILE
+        error_check "Failed to install Dasharo EC firmware"
+	print_ok "Successfully installed Dasharo EC firmware"
       fi
     fi
+
+    return 0
   fi
 
-  if [ "$CAN_INSTALL_BIOS" == "true" ]; then
-    cbfstool "$BIOS_UPDATE_FILE" extract -r COREBOOT -n config -f "$BIOS_UPDATE_CONFIG_FILE"
-    grep "CONFIG_VBOOT=y" "$BIOS_UPDATE_CONFIG_FILE"
-    HAVE_VBOOT="$?"
+  # Must not get here:
+  return 1
+}
 
-    if [ "$NEED_ROMHOLE_MIGRATION" = "true" ]; then
-      romhole_migration
-    fi
+firmware_pre_updating_routine(){
+# This function only separates some code from deployment code, so to make clear
+# where is deployment code, and what should be executed before it:
+  check_flash_lock
+
+  if [ "$HAVE_EC" == "true" ]; then
+    check_for_opensource_firmware
+    error_check "Device does not have Dasharo EC firmware - cannot continue update!"
+  fi
+
+  if [ "$NEED_SMMSTORE_MIGRATION" = "true" ]; then
+    smmstore_migration
+  fi
 
-    if [ "$NEED_BOOTSPLASH_MIGRATION" = "true" ]; then
-      bootsplash_migration
+  if [ "$NEED_BOOTSPLASH_MIGRATION" = "true" ]; then
+    bootsplash_migration
+  fi
+
+  $CBFSTOOL read_bios_conffile_mock "$BIOS_UPDATE_FILE" extract -r COREBOOT -n config -f "$BIOS_UPDATE_CONFIG_FILE"
+  grep -q "CONFIG_VBOOT=y" "$BIOS_UPDATE_CONFIG_FILE"
+  HAVE_VBOOT="$?"
+
+  check_intel_regions
+  check_blobs_in_binary $BIOS_UPDATE_FILE
+  check_if_me_disabled
+  set_flashrom_update_params $BIOS_UPDATE_FILE
+  set_intel_regions_update_params "-N --ifd"
+  check_vboot_keys
+
+  return 0
+}
+
+firmware_pre_installation_routine(){
```

Comment:
You don't use this function anywhere

Reply:
Thanks, very important catch! Fixed [here](https://github.com/Dasharo/dts-scripts/compare/a29856f198488767178cbe8e15853f96def1fb8b..ed6c6089caa0e541f15150b15a79a7e26c6ae365).